dataactcore/scripts/pullFPDSData.py

```diff
@@ -0,0 +1,1124 @@
+
+    # clear out potentially excel-breaking whitespace from specific fields
+    free_fields = ["award_description", "vendor_doing_as_business_n", "legal_entity_address_line1",
+                   "legal_entity_address_line2", "legal_entity_address_line3", "ultimate_parent_legal_enti",
+                   "awardee_or_recipient_legal", "other_statutory_authority"]
+    for field in free_fields:
+        if obj[field]:
+            obj[field] = re.sub('\s', ' ', obj[field])
+
+    return obj
+
+
+def process_delete_data(data, atom_type):
+    obj = {}
+
+    # get all values that make up unique key
+    if atom_type == "award":
+        try:
+            obj['piid'] = extract_text(data['awardID']['awardContractID']['PIID'])
+        except (KeyError, TypeError):
+            obj['piid'] = None
+
+        try:
+            obj['agency_id'] = extract_text(data['awardID']['awardContractID']['agencyID'])
+        except (KeyError, TypeError):
+            obj['agency_id'] = None
+
+        try:
+            obj['award_modification_amendme'] = extract_text(data['awardID']['awardContractID']['modNumber'])
+        except (KeyError, TypeError):
+            obj['award_modification_amendme'] = None
+
+        try:
+            obj['parent_award_id'] = extract_text(data['awardID']['referencedIDVID']['PIID'])
+        except (KeyError, TypeError):
+            obj['parent_award_id'] = None
+
+        try:
+            obj['referenced_idv_agency_iden'] = extract_text(data['awardID']['referencedIDVID']['agencyID'])
+        except (KeyError, TypeError):
+            obj['referenced_idv_agency_iden'] = None
+
+        try:
+            obj['transaction_number'] = extract_text(data['awardID']['awardContractID']['transactionNumber'])
+        except (KeyError, TypeError):
+            obj['transaction_number'] = None
+    else:
+        try:
+            obj['piid'] = extract_text(data['contractID']['IDVID']['PIID'])
+        except (KeyError, TypeError):
+            obj['piid'] = None
+
+        try:
+            obj['agency_id'] = extract_text(data['contractID']['IDVID']['agencyID'])
+        except (KeyError, TypeError):
+            obj['agency_id'] = None
+
+        try:
+            obj['award_modification_amendme'] = extract_text(data['contractID']['IDVID']['modNumber'])
+        except (KeyError, TypeError):
+            obj['award_modification_amendme'] = None
+
+        try:
+            obj['parent_award_id'] = extract_text(data['contractID']['referencedIDVID']['PIID'])
+        except (KeyError, TypeError):
+            obj['parent_award_id'] = None
+
+        try:
+            obj['referenced_idv_agency_iden'] = extract_text(data['contractID']['referencedIDVID']['agencyID'])
+        except (KeyError, TypeError):
+            obj['referenced_idv_agency_iden'] = None
+
+        # not in IDV feed, just set it to None
+        obj['transaction_number'] = None
+
+    return obj
+
+
+def process_and_add(data, contract_type, sess, last_run=None):
```

Comment:
Some additional comments here about what checking `last_run` means would be useful to folks looking at this for the first time